jwst/exp_to_source/exp_to_source.py

```diff
@@ -40,8 +40,20 @@ def exp_to_source(inputs):
             log.debug(f'Copying source {slit.source_id}')
             result_slit = result[str(slit.source_id)]
             result_slit.exposures.append(slit)
-            merge_tree(result_slit.exposures[-1].meta.instance, exposure.meta.instance)
+            # store values for later use (after merge_tree)
+            slit_bunit = slit.meta.bunit_data
+            slit_bunit_err = slit.meta.bunit_err
+            slit_model = slit.meta.model_type
+            slit_wcs = slit.meta.wcs
+
+            # exposure.meta.bunit_data and bunit_err does not exist
+            # before calling merge_tree save these values
 
+            merge_tree(result_slit.exposures[-1].meta.instance, exposure.meta.instance)
+            result_slit.exposures[-1].meta.bunit_data = slit_bunit
+            result_slit.exposures[-1].meta.bunit_err = slit_bunit_err
+            result_slit.exposures[-1].meta.model_type = slit_model
```

Comment:
Do the `model_type` and `wcs` attributes also not exist before calling `merge_tree` and hence have to be copied, like `bunit_data` and `bunit_err`? If so, they should be included in the description in the comment, or given their own description as to why they're being saved and copied over.

Reply:
I was thinking the worry was that model_type and wcs for the slit model would exist but there values would be overwritten by merge_tree. Maybe I should not touch the wcs since I am not sure and that could break so many down stream routines if I could in the incorrect value.
For model type - before merge_tree (in exp_to_source) the is is 'SlitModel' after merge_tree it is MultiSlitModel.
Which might be ok or even preferred depending on what downstream routines expect. I might remove these two
values being saved and then over writing the values because we do not have an issue with either of these down stream so I don't want to break anything

Reply:
Ah, now that I read Tyler's comment above I understand why you added the saving/copying of `model_type` and `wcs`, in addition to the bunit stuff. So I'm OK with all this.

Reply:
To be clear, my other comment is with respect to the `meta.wcsinfo` entries, not the `meta.wcs` WCS object itself. That was not being overwritten (from what I recall from testing), but the WCS keywords stored in `meta.wcsinfo` were. This will cause a mismatch between the WCS and the information in the wcsinfo if the override in `merge_tree` is allowed to propagate, as it currently is.

Reply:
@tapastro Thanks for explaining that. I was confused. I now see that meta.wcsinfo is getting overwritten and I have  added that to the values that are saved.